src/Native/Unix/System.Native/pal_io.cpp

```diff
@@ -1171,3 +1171,22 @@ extern "C" char* SystemNative_RealPath(const char* path)
     assert(path != nullptr);
     return realpath(path, nullptr);
 }
+
+extern "C" int32_t SystemNative_LockFileRegion(int fd, int64_t offset, int64_t length, short lockType)
```

Comment:
@stephentoub updated this with your feedback. I made the locktype enum a short that we can pass instead of the bool. I am curious if I should explicitly make this an `int16_t `.